src/ompl/base/objectives/src/PathLengthOptimizationObjective.cpp

```diff
@@ -62,10 +64,13 @@ ompl::base::Cost ompl::base::PathLengthOptimizationObjective::motionCostHeuristi
     return motionCost(s1, s2);
 }
 
-#if OMPL_HAVE_EIGEN3
 ompl::base::InformedSamplerPtr ompl::base::PathLengthOptimizationObjective::allocInformedStateSampler(const ProblemDefinitionPtr probDefn, unsigned int maxNumberCalls) const
 {
-    // Make the direct path-length informed sampler and return
+    // Make the direct path-length informed sampler and return. If OMPL was compiled with Eigen, a direct version is available, if not a rejection-based technique can be used
+#if OMPL_HAVE_EIGEN3
     return boost::make_shared<PathLengthDirectInfSampler>(probDefn, maxNumberCalls);
-}
+#else
+    OMPL_INFORM("%s: Direct sampling of the path-length objective requires Eigen. As this version of OMPL was compiled without Eigen support, the informed sampling is defaulting to rejection sampling.", description_.c_str());
+    return boost::make_shared<RejectionInfSampler>(probDefn, maxNumberCalls);
```

Comment:
So my thought is that it is completely valid to use rejection sampling for path length if a user is in a situation where they **can't** have Eigen... I do agree; however, that if they **could** have Eigen and simply **don't**, it'd be better to forcefully suggest they install it. :wink:

So how about a preprocessor directive to acknowledge they understand and want to use rejection sampling? Then the else block for `OMPL_HAVE_EIGEN3` would either create a rejection sampler or throw an error describing the options (acknowledge this is what you want to do, or install Eigen).

I'm thinking something like they do in `boost::ublas`, e.g., requiring the user to do something like

    #define OMPL_ALLOW_REJECTION_SAMPLING_FOR_PATH_LENGTH_OBJECTIVE

in their file.

I believe this will require moving the implementation of this function to the header though, as it's compile-time. Is that a problem?